src/RandomEnchants.cpp

```diff
@@ -6,137 +6,140 @@
+        uint32 slotEnch[3] = { 0, 1, 5 };
+
+        // Fetching the configuration values as float
+        float enchantChance1 = sConfigMgr->GetOption<float>("RandomEnchants.EnchantChance1", 70.0f);
+        float enchantChance2 = sConfigMgr->GetOption<float>("RandomEnchants.EnchantChance2", 65.0f);
+        float enchantChance3 = sConfigMgr->GetOption<float>("RandomEnchants.EnchantChance3", 60.0f);
+
+        if (rand_chance() < enchantChance1)
+            slotRand[0] = getRandEnchantment(item);
+        if (slotRand[0] != -1 && rand_chance() < enchantChance2)
+            slotRand[1] = getRandEnchantment(item);
+        if (slotRand[1] != -1 && rand_chance() < enchantChance3)
+            slotRand[2] = getRandEnchantment(item);
+
+        for (int i = 0; i < 3; i++) {
+            if (slotRand[i] != -1) {
+                if (sSpellItemEnchantmentStore.LookupEntry(slotRand[i])) { //Make sure enchantment id exists
+                    player->ApplyEnchantment(item, EnchantmentSlot(slotEnch[i]), false);
+                    item->SetEnchantment(EnchantmentSlot(slotEnch[i]), slotRand[i], 0, 0);
+                    player->ApplyEnchantment(item, EnchantmentSlot(slotEnch[i]), true);
+                }
+            }
+        }
+        ChatHandler chathandle = ChatHandler(player->GetSession());
+        if (slotRand[2] != -1)
+            chathandle.PSendSysMessage("Newly Acquired |cffFF0000 %s |rhas received|cffFF0000 3 |rrandom enchantments!", item->GetTemplate()->Name1.c_str());
+        else if (slotRand[1] != -1)
+            chathandle.PSendSysMessage("Newly Acquired |cffFF0000 %s |rhas received|cffFF0000 2 |rrandom enchantments!", item->GetTemplate()->Name1.c_str());
+        else if (slotRand[0] != -1)
+            chathandle.PSendSysMessage("Newly Acquired |cffFF0000 %s |rhas received|cffFF0000 1 |rrandom enchantment!", item->GetTemplate()->Name1.c_str());
+    }
+
+
+    int getRandEnchantment(Item* item) {
+        uint32 Class = item->GetTemplate()->Class;
+        std::string ClassQueryString = "";
+        switch (Class) {
+        case 2:
+            ClassQueryString = "WEAPON";
+            break;
+        case 4:
+            ClassQueryString = "ARMOR";
+            break;
+        }
+        if (ClassQueryString == "")
+            return -1;
+        uint32 Quality = item->GetTemplate()->Quality;
+        int rarityRoll = -1;
+        switch (Quality) {
+        case 0: //grey
+            rarityRoll = rand_norm() * 25;
+            break;
+        case 1: //white
+            rarityRoll = rand_norm() * 50;
+            break;
+        case 2: //green
+            rarityRoll = 45 + (rand_norm() * 20);
+            break;
+        case 3: //blue
+            rarityRoll = 65 + (rand_norm() * 15);
+            break;
+        case 4: //purple
+            rarityRoll = 80 + (rand_norm() * 14);
+            break;
+        case 5: //orange
+            rarityRoll = 93;
+            break;
```

Comment:
please, create an enum instead of using `case 0: // grey` use `case GREY:` where GREY is the variable in the enum